src/GovUk.Education.ExploreEducationStatistics.Notifier/Functions/ReleaseNotifier.cs

```diff
@@ -0,0 +1,178 @@
+
+        foreach (var supersededPublication in msg.SupersededPublications)
+        {
+            var releaseSupersededPubSubsQuery = new TableQuery<SubscriptionEntity>()
+                .Where(TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.Equal,
+                    supersededPublication.Id.ToString()));
+            var supersededPublicationSubscriberEmails = await GetEmails(
+                table, releaseSupersededPubSubsQuery, logger);
+
+            var numSupersededSubscriberEmailsSent = 0;
+
+            foreach (var email in supersededPublicationSubscriberEmails)
+            {
+                if (sentEmails.Contains(email))
+                {
+                    continue;
+                }
+
+                numSupersededSubscriberEmailsSent++;
+                sentEmails.Add(email);
+
+                var unsubscribeToken =
+                    _tokenService.GenerateToken(tokenSecretKey, email, DateTime.UtcNow.AddYears(1));
+                var values = new Dictionary<string, dynamic>
+                {
+                    // Use values from the queue just in case the name or slug of the publication changes
+                    { "publication_name", msg.PublicationName },
+                    { "release_name", msg.ReleaseName },
+                    {
+                        "release_link",
+                        $"{webApplicationBaseUrl}find-statistics/{msg.PublicationSlug}/{msg.ReleaseSlug}"
+                    },
+                    {
+                        // NOTE: update_note not used by original release email template
+                        "update_note", msg.UpdateNote
+                    },
+                    { "unsubscribe_link", $"{baseUrl}{supersededPublication.Id}/unsubscribe/{unsubscribeToken}" },
+                    { "superseded_publication_title", supersededPublication.Title },
+                };
+
+                _emailService.SendEmail(notificationClient, email, newReleaseSupersededSubscribersEmailTemplateId, values);
+            }
+            logger.LogInformation("Emailed {NumSupersededPublicationEmailsSent} subscribers from a superseded publication",
+                numSupersededSubscriberEmailsSent);
+        }
+        logger.LogInformation("Sent {TotalNumEmailsSent} emails in total to subscribers",
+            sentEmails.Count);
+    }
+
+    private static async Task<List<string>> GetEmails(
+        CloudTable table,
+        TableQuery<SubscriptionEntity> query,
+        ILogger logger)
+    {
+        var emails = new List<string>();
+
+        TableContinuationToken? token = null;
+        do
+        {
+            var resultSegment =
+                await table.ExecuteQuerySegmentedAsync(query, token);
+            token = resultSegment.ContinuationToken;
+
+            logger.LogInformation("Emailing {SubscriberCount} subscribers",
+                resultSegment.Results.Count);
```

Comment:
You don't need this here so this method probably doesn't neeed `logger` at all anymore.